lib/diagnostics_channel.js

```diff
@@ -138,10 +226,171 @@ function hasSubscribers(name) {
   return channel.hasSubscribers;
 }
 
+const traceEvents = [
+  'start',
+  'end',
+  'asyncStart',
+  'asyncEnd',
+  'error',
+];
+
+function assertChannel(value, name) {
+  if (!(value instanceof Channel)) {
+    throw new ERR_INVALID_ARG_TYPE(name, ['Channel'], value);
+  }
+}
+
+class TracingChannel {
+  constructor(nameOrChannels) {
+    if (typeof nameOrChannels === 'string') {
+      this.start = channel(`tracing:${nameOrChannels}:start`);
+      this.end = channel(`tracing:${nameOrChannels}:end`);
+      this.asyncStart = channel(`tracing:${nameOrChannels}:asyncStart`);
+      this.asyncEnd = channel(`tracing:${nameOrChannels}:asyncEnd`);
+      this.error = channel(`tracing:${nameOrChannels}:error`);
+    } else if (typeof nameOrChannels === 'object') {
+      const { start, end, asyncStart, asyncEnd, error } = nameOrChannels;
```

Comment:
Is there any way to add additional channels, or are those expected to be defined externally with a custom context object passed manually?

Reply:
It's expected to be done externally in this design, though you _can_ share the context object by just using the same context object you give the `trace*` functions. Some helper to support additional channels could be considered too, though I would recommend we consider that as a follow-up if at all.

Reply:
This could definitely be added later, I just thought that otherwise it will result in exposing some of the implementation details.